### PR TITLE
Add encryption endpoint in supervisor frontend

### PR DESCRIPTION
**Pull request by mustard-mh**
## Related Issue(s)

Fixes #
Relates https://github.com/gitpod-io/security/issues/121

## How to test

- Open workspace in preview env with latest browser code
- Check localStorage if `credentials.providers` is encrypted

**compatible#1**
- Close workspace and change default IDE to stable browser code
- Re-open workspace, check if everything is alright (compatible)

**compatible#2 old workspace reopen**
- Copy credentials.provider from production and set
- Re-open workspace use stable browser code
- Check if it encrypted again and the decrypted value is the same with just copied value

----
✅ Use latest browser code to start a workspace in preview env confirmed that it's encrypted, change it's value (add copy and change service name) refresh still exists, restart workspace still exists
|Change Value|Restart Workspace|
|:-:|:-:|
|SCR-20230215-ux6|SCR-20230215-uxv|

✅ Start with stable browser code, credentials should not be encrypted, change it's value and stop workspace, restart with latest code, it should work
|Change Value|Restart Workspace with IDE Changed|
|:-:|:-:|
|image|image|

✅ Override supervisor-frontend script to not set credentailsToken (mock server rollback), it should failed and use plain text again (user may need to re-auth), add new service name and refresh, it's still here with plain text
|Override|Plain Text|Add One|
|:-:|:-:|:-:|
|image|image|image|

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`
- [ ] /werft publish-to-npm

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@mustard-mh: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by mustard-mh** [comment]
/hold

To remove debug commit

**Comment by mustard-mh** [comment]
I have tested in preview env, results see PR description 🙏

File: components/gitpod-protocol/src/typings/globals.ts
Comment (akosyakov) [inline components/gitpod-protocol/src/typings/globals.ts:11]: Should not we make @gitpod-io/team-gitpod owners to avoid bothering webapp each time?  @geropl @easyCZ fine?

Comment (mustard-mh) [reply components/gitpod-protocol/src/typings/globals.ts:11]: CODEOWNERS changed in this PR 🙏

File: components/supervisor/frontend/src/shared/frontend-dashboard-service.ts
Comment (jeanp413) [inline components/supervisor/frontend/src/shared/frontend-dashboard-service.ts:69]: do we have a preference for the encryption algorithm? cc @geropl @akosyakov @securitymirco

**Review by jeanp413 — APPROVED** [review]

**Review by svenefftinge — APPROVED** [review]

**Pull request merged**
